src/main/java/software/amazon/awssdk/crt/utils/StringUtils.java

```diff
@@ -23,4 +23,48 @@ public static String join(CharSequence delimiter, Iterable<? extends CharSequenc
         }
         return sb.toString();
     }
+
+    /**
+     * Encode a byte[] array into a Base64 String in pure Java
+     */
+    private final static char[] BASE64_VALID_CHARACTERS = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/".toCharArray();
+    public static String simpleBase64Encode(byte[] data) {
```

Comment:
Can we add some tests for this ala the tests for the common impl: https://github.com/awslabs/aws-c-common/blob/main/tests/encoding_test.c#L395

Reply:
The tests in the common impl have been added! I skipped all of the decoder tests, as this function just encodes and there is no decoder for now since we only need encoding for the Java keystore.